src/lib/util/calculateGearLostOnDeathWilderness.ts

```diff
@@ -0,0 +1,221 @@
+	12848: [itemID('Granite maul'), itemID('Granite clamp')],
+	24227: [24225, itemID('Granite clamp')],
+	[itemID('Abyssal tentacle')]: [itemID('Abyssal whip'), itemID('Kraken tentacle')]
+};
+
+export const lockedItems = resolveItems([
+	'Rune pouch (l)',
+	'Fire cape (l)',
+	'Infernal cape (l)',
+	"Ava's assembler (l)",
+	'Imbued guthix cape (l)',
+	'Imbued saradomin cape (l)',
+	'Imbued zamorak cape (l)',
+	'Fire max cape (l)',
+	'Infernal max cape (l)',
+	'Assembler max cape (l)',
+	'Imbued guthix max cape (l)',
+	'Imbued saradomin max cape (l)',
+	'Imbued zamorak max cape (l)',
+	'Bronze defender (l)',
+	'Iron defender (l)',
+	'Steel defender (l)',
+	'Black defender (l)',
+	'Mithril defender (l)',
+	'Adamant defender (l)',
+	'Rune defender (l)',
+	'Dragon defender (l)',
+	'Avernic defender (l)',
+	'Void melee helm (l)',
+	'Void mage helm (l)',
+	'Void ranger helm (l)',
+	'Void knight top (l)',
+	'Elite void top (l)',
+	'Void knight robe (l)',
+	'Elite void robe (l)',
+	'Void knight gloves (l)',
+	// Decorative armour gold
+	24158,
+	24159,
+	24162,
+	// Decorative magic armour
+	24163,
+	24164,
+	24165,
+	// Decorative ranged armour
+	24166,
+	24167,
+	24168,
+	'Guthix halo (l)',
+	'Saradomin halo (l)',
+	'Zamorak halo (l)',
+	'Armadyl halo (l)',
+	'Bandos halo (l)',
+	'Seren halo (l)',
+	'Ancient halo (l)',
+	'Brassica halo (l)',
+	'Fighter hat (l)',
+	'Ranger hat (l)',
+	'Healer hat (l)',
+	'Fighter torso (l)',
+	'Penance skirt (l)'
+]);
+
+export default function calculateGearLostOnDeathWilderness(
+	options: {
+		gear: GearSetup | undefined;
+		skulled: boolean;
+		after20wilderness: boolean;
+		smited: boolean;
+		protectItem: boolean;
+	} = {
+		gear: undefined,
+		skulled: false,
+		smited: false,
+		protectItem: false,
+		after20wilderness: false
+	}
+) {
+	if (!options.gear) {
+		return false;
+	}
+
+	// 1 - Duplicate user gear
+	let userGear = { ...options.gear };
+	let removableItems: { slot: EquipmentSlot; sorter: number; originalItem: Item }[] = [];
+
+	// 2 - Swap user gear to the correct gear for death calculations
+	for (const [_slot, _data] of objectEntries(userGear)) {
+		if (!_data || _data.item === undefined) continue;
+		const originalItem = _data.item;
+		let replacedGear = [_data.item];
+		if (gearSwap[_data.item]) replacedGear = gearSwap[_data.item];
+
+		removableItems.push({
+			slot: _slot,
+			sorter: replacedGear
+				.map(_i => {
+					const i = getOSItem(_i);
+					if (lockedItems.includes(i.id)) return -1;
+					// Get the hichest value for the item, be it protection value (cost), ge price or high alch
+					return Math.max(i.price, i.cost, i.highalch);
+				})
+				.reduce((sum, current) => sum + current, 0),
+			originalItem: getOSItem(originalItem)
+		});
+	}
+
+	// 3 - Sort gear
+	const sortedGear = removableItems.sort((a, b) => b.sorter - a.sorter);
+
+	// 4 - Replicate the sorting to the main gear
+	const lostItemsBank = new Bank();
+	for (const { slot, originalItem } of sortedGear.slice(
```

Comment:
The ternary here for numSavedItems feels unnecessary and hard to read. Perhaps pull it out of the slice and instead define it beforehand?